src/Nethermind/Nethermind.DataMarketplace.Consumers.Infrastructure/NdmConsumersModule.cs

```diff
@@ -150,15 +150,17 @@ public void Init()
                     sessionRepository = new ConsumerSessionInMemoryRepository();
                     break;
                 default:
-                    depositRepository = new DepositDetailsRocksRepository(rocksDbProvider.DepositsDb,
+                    var dbInitializer = new ConsumerNdmDbInitializer(_api.DbProvider, ndmConfig, _api.RocksDbFactory, _api.MemDbFactory);
+                    await dbInitializer.Init();
+                    depositRepository = new DepositDetailsRocksRepository(_api.DbProvider.GetDb<IDb>(ConsumerNdmDbConsts.DepositsDbName),
                         depositDetailsRlpDecoder);
                     depositApprovalRepository = new ConsumerDepositApprovalRocksRepository(
-                        rocksDbProvider.ConsumerDepositApprovalsDb, depositApprovalRlpDecoder);
-                    providerRepository = new ProviderRocksRepository(rocksDbProvider.DepositsDb,
+                        _api.DbProvider.GetDb<IDb>(ConsumerNdmDbConsts.ConsumerDepositApprovalsDbName), depositApprovalRlpDecoder);
```

Comment:
_api.Db(name) => _api.GetDb<IDb>(...)
_api.Db<T>(name) => _api.DbProvider.GetDb<T>(...)
===
similar as for configs maybe?